File: backend/src/main/java/com/anti/entity/News.java
package com.anti.entity;

import com.baomidou.mybatisplus.annotation.*;
import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Data;

import java.io.Serializable;
import java.time.LocalDateTime;

/**
 * 资讯实体类
 */
@Data
@TableName("news")
public class News implements Serializable {

    @TableId(type = IdType.AUTO)
    private Long id;

    private String title;

    private String content;

    private String summary;

    @TableField("cover_image")
    private String coverImage;

    @TableField("category_id")
    private Long categoryId;

    @TableField("author_id")
    private Long authorId;

    @TableField("news_type")
    private String newsType;

    @TableField("is_top")
    private Integer isTop;

    @TableField("is_mandatory")
    private Integer isMandatory;

    @TableField("view_count")
    private Integer viewCount;

    private Integer status;

    @TableField("publish_time")
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime publishTime;

    @TableField(fill = FieldFill.INSERT)
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime createTime;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime updateTime;

    @TableField(exist = false)
    private static final long serialVersionUID = 1L;

    @TableField(exist = false)
    private String categoryName;

    @TableField(exist = false)
    private String authorName;

    @TableField(exist = false)
    private Long likeCount;

    @TableField(exist = false)
    private Boolean isLiked;
}

File: frontend/src/api/news.ts
import request from '@/utils/request'

export interface News {
  id: number
  title: string
  content: string
  summary: string
  coverImage: string
  categoryId: number
  categoryName: string
  authorId: number
  authorName: string
  newsType: 'news' | 'warning' | 'policy'
  isTop: number
  isMandatory: number
  viewCount: number
  likeCount: number
  isLiked: boolean
  status: number
  publishTime: string
  createTime: string
}

export interface NewsCategory {
  id: number
  name: string
  parentId: number
  sortOrder: number
  createTime: string
}

export interface PageResult<T> {
  records: T[]
  total: number
  size: number
  current: number
}

export const getNewsPage = (params: {
  pageNum: number
  pageSize: number
  categoryId?: number
  newsType?: string
  keyword?: string
}) => {
  return request.get<{ data: PageResult<News> }>('/news/page', { params })
}

export const getNewsDetail = (id: number) => {
  return request.get<{ data: News }>(`/news/${id}`)
}

export const likeNews = (id: number) => {
  return request.post<{ data: boolean }>(`/news/${id}/like`)
}

export const unlikeNews = (id: number) => {
  return request.delete<{ data: boolean }>(`/news/${id}/like`)
}

export const viewNews = (id: number, stayDuration?: number) => {
  return request.post(`/news/${id}/view`, { stayDuration })
}

export const getBrowseHistory = (params: { pageNum: number; pageSize: number }) => {
  return request.get<{ data: PageResult<News> }>('/news/browse/history', { params })
}

export const getCategories = () => {
  return request.get<{ data: NewsCategory[] }>('/news/category/list')
}
